lib/documents.js

```diff
@@ -11,6 +11,7 @@ import canonicalize from 'canonicalize';
 import {Cipher} from 'minimal-cipher';
 import {promisify} from 'util';
 import {createKeyResolver} from './keyResolver.js';
+import {updateBatches} from './tokens.js';
```

Comment:
This creates a circular dependency. We need to fix it. I think the way to fix it -- and to make the error I mention down below make a bit more sense is to change the `updateBatches` flag to a flag that indicates that we require that the registration NOT be new. We can use the existing `newRegistration` flag to handle it having three possible values: `true`, `false`, or `undefined`. When we want to update an existing registration, we set it to `false` and throw if there is no existing registration. Then, if an existing registration is successfully updated, we can call `updateBatches` (which we may also want to rename because it is specifically changing the `minAssuranceForResolution`).

```suggestion
```